lib/absinthe/phase/document/variables.ex

```diff
@@ -76,4 +72,19 @@ defmodule Absinthe.Phase.Document.Variables do
     {node, acc}
   end
 
+  defp get(map, key, default) do
+    if Map.has_key?(map, key) do
+      Map.get(map, key) |> Blueprint.Input.parse
+    else
+      case {Map.get(map, key), default} do
```

Comment:
Ah, I see now. It's that last case (`nil` stored as an explicit Map value) that I wasn't considering. Thanks for the explanation!